src/aws-cpp-sdk-core/include/smithy/identity/identity/AwsCredentialIdentity.h

```diff
@@ -12,20 +12,26 @@ namespace smithy {
         AwsCredentialIdentity(const Aws::String& accessKeyId,
                               const Aws::String& secretAccessKey,
                               const Aws::Crt::Optional<Aws::String>& sessionToken,
-                              const Aws::Crt::Optional<AwsIdentity::DateTime>& expiration)
-            : m_accessKeyId(accessKeyId), m_secretAccessKey(secretAccessKey),
-              m_sessionToken(sessionToken), m_expiration(expiration) {}
+                              const Aws::Crt::Optional<AwsIdentity::DateTime>& expiration,
+                              const Aws::Crt::Optional<Aws::String>& accountId)
+            : m_accessKeyId(accessKeyId),
+              m_secretAccessKey(secretAccessKey),
+              m_sessionToken(sessionToken),
+              m_expiration(expiration),
+              m_accountId({accountId}) {}
```

Comment:
I think you are accidentally creating a temporary r-value copy here and then move-construct m_accountId

I think `m_accountId(accountId)` should just work.